Jenkinsfile

```diff
@@ -8,72 +8,72 @@ pipeline {
-        // }
+        stage('Catkin build on ROS workspace') {
+          agent {
+            docker {
+              image 'px4io/px4-dev-ros-melodic:2019-03-08'
+              args '-e CCACHE_BASEDIR=$WORKSPACE -v ${CCACHE_DIR}:${CCACHE_DIR}:rw -e HOME=$WORKSPACE'
+            }
+          }
+          steps {
+            sh 'ls -l'
+            sh '''#!/bin/bash -l
+              echo $0;
+              mkdir -p catkin_ws/src;
+              cd catkin_ws;
+              git -C ${WORKSPACE}/catkin_ws/src/Firmware submodule update --init --recursive --force Tools/sitl_gazebo
+              git clone --recursive ${WORKSPACE}/catkin_ws/src/Firmware/Tools/sitl_gazebo src/mavlink_sitl_gazebo;
+              source /opt/ros/melodic/setup.bash;
+              catkin init;
+              catkin build -j$(nproc) -l$(nproc);
+            '''
+            // test if the binary was correctly installed and runs using 'mavros_posix_silt.launch'
+            sh '''#!/bin/bash -l
+              echo $0;
+              source catkin_ws/devel/setup.bash;
+              rostest px4 pub_test.launch;
+            '''
+          }
+          post {
+            always {
+              sh 'rm -rf catkin_ws'
+            }
+          }
+          options {
+            checkoutToSubdirectory('catkin_ws/src/Firmware')
+          }
+        }
 
-        // stage('Colcon build on ROS2 workspace') {
-        //   agent {
-        //     docker {
-        //       image 'px4io/px4-dev-ros2-bouncy:2019-03-08'
-        //       args '-e CCACHE_BASEDIR=$WORKSPACE -v ${CCACHE_DIR}:${CCACHE_DIR}:rw -e HOME=$WORKSPACE'
-        //     }
-        //   }
-        //   steps {
-        //     sh 'ls -l'
-        //     sh '''#!/bin/bash -l
-        //       echo $0;
-        //       unset ROS_DISTRO;
-        //       mkdir -p colcon_ws/src;
-        //       cd colcon_ws;
-        //       git -C ${WORKSPACE}/colcon_ws/src/Firmware submodule update --init --recursive --force Tools/sitl_gazebo
-        //       git clone --recursive ${WORKSPACE}/colcon_ws/src/Firmware/Tools/sitl_gazebo src/mavlink_sitl_gazebo;
-        //       source /opt/ros/bouncy/setup.sh;
-        //       source /opt/ros/melodic/setup.sh;
-        //       colcon build --event-handlers console_direct+ --symlink-install;
-        //     '''
-        //   }
-        //   post {
-        //     always {
-        //       sh 'rm -rf colcon_ws'
-        //     }
-        //   }
-        //   options {
-        //     checkoutToSubdirectory('colcon_ws/src/Firmware')
-        //   }
-        // }
+        stage('Colcon build on ROS2 workspace') {
+          agent {
+            docker {
+              image 'px4io/px4-dev-ros2-bouncy:2019-03-08'
+              args '-e CCACHE_BASEDIR=$WORKSPACE -v ${CCACHE_DIR}:${CCACHE_DIR}:rw -e HOME=$WORKSPACE'
+            }
+          }
+          steps {
+            sh 'ls -l'
+            sh '''#!/bin/bash -l
+              echo $0;
+              unset ROS_DISTRO;
```

Comment:
Why is this needed?

Reply:
On a environment with multiple ROS versions installed, if one previously have sourced a certain ROS distro env setup, different from the one you are sourcing now, there will be a warning message similar to `ROS_DISTRO was set to 'crystal' before. Please make sure that the environment does not mix paths from different distributions.`. This just avoids that this message spams the console, as we first unset the `ROS_DISTRO` env variable so there are no conflicts.